api/v1alpha1/nfspvc_webhook.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var c client.Client
+
+const (
+	UpdateNfsPvcError      = "forbidden: NFSPVC spec is immutable after creation"
+	PVCAlreadyExists       = "a PVC of this name already exists in the namespace. Please rename your NFSPVC"
+	InvalidAccessModeError = "forbidden: only the following AccessModes are permitted"
+)
+
+var supportedAccessModes = sets.New(
+	corev1.ReadWriteOnce,
+	corev1.ReadOnlyMany,
+	corev1.ReadWriteMany,
+	corev1.ReadWriteOncePod)
```

Comment:
Think the end parenthesis needs to be in a newline? How does `go fmt` handle it?